modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -840,6 +840,79 @@ TEST(Imgcodecs_Tiff, readWrite_predictor)
     }
 }
 
+// See https://github.com/opencv/opencv/issues/23416
+typedef std::pair<int,bool> Imgcodes_Tiff_TypeAndComp;
+typedef testing::TestWithParam< Imgcodes_Tiff_TypeAndComp > Imgcodecs_Tiff_Types;
+
+TEST_P(Imgcodecs_Tiff_Types, readWrite_alltypes)
+{
+    const int types = get<0>(GetParam());
+    const bool isCompAvailable = get<1>(GetParam());
+
+    const Mat img = cv::Mat::zeros( 120, 160, types );
+    ASSERT_FALSE(img.empty());
+    {
+        // Add noise to test compression.
+        cv::Mat roi = cv::Mat(img, cv::Rect(0, 0, img.cols, img.rows/2));
+        cv::randu(roi, cv::Scalar(0), cv::Scalar(256));
+    }
+
+    std::vector<uchar> bufLZW;
+    {
+        std::vector<int> params;
+        params.push_back(IMWRITE_TIFF_COMPRESSION);
+        params.push_back(COMPRESSION_LZW);
+
+        EXPECT_NO_THROW(cv::imencode(".tiff", img, bufLZW, params));
+    }
+
+    const Mat imgLZW = cv::imdecode( bufLZW, IMREAD_UNCHANGED);
+    {
+        ASSERT_FALSE(imgLZW.empty());
+        ASSERT_EQ(imgLZW.type(), img.type());
+        ASSERT_EQ(imgLZW.size(), img.size());
+    }
+
+    std::vector<uchar> bufRAW;
+    {
+        std::vector<int> params;
+        params.push_back(IMWRITE_TIFF_COMPRESSION);
+        params.push_back(COMPRESSION_NONE);
+
+        EXPECT_NO_THROW(cv::imencode(".tiff", img, bufRAW, params));
+    }
+
+    const Mat imgRAW = cv::imdecode( bufRAW, IMREAD_UNCHANGED);
+    {
+        ASSERT_FALSE(imgRAW.empty());
```

Comment:
Size and type check are enough. Empty check is redundant.

Reply:
Some test codes use `src.empty()` to check the result of imread(). 

https://github.com/opencv/opencv/blob/725e440d278aca07d35a5e8963ef990572b07316/modules/imgcodecs/test/test_tiff.cpp#L570-L576

However I found another way to test imread. I think it is more user-friendly style for others. In New PR, I replaced as this style.

https://github.com/opencv/opencv/blob/725e440d278aca07d35a5e8963ef990572b07316/modules/imgcodecs/test/test_tiff.cpp#L683